libraries/chain/block_state.cpp

```diff
@@ -29,6 +35,7 @@ block_state::block_state(const block_header_state& bhs, deque<transaction_metada
 block_state::block_state(const block_state_legacy& bsp) {
    block_header_state::id = bsp.id();
    header = bsp.header;
+   core.last_final_block_num = bsp.block_num();
```

Comment:
Probably add a `todo` because this will change when we dont transition immediately when executing set_finalizers.